### PR TITLE
Fix installation

### PR DESCRIPTION
**Pull request by jdahm**
We can merge this before #306 in LLNL/blt, but the submodule should be updated when that happens.

### CODE REVIEW

**Comment by jdahm** [comment]
This is ready to go!

**Comment by mcfadden8** [comment]
@jdahm, Rather than (or in addition to) trying to decipher what the installation directory will look like by looking at the code differences, can you please provide a tree view of what the install directory is intended to look like?  It would be good to see what it looks like for installations that do and don't include tools.  Are there other options?

Thanks,

Marty

**Comment by jdahm** [comment]
@mcfadden8 it creates the following directories under `install/`, and puts files there

```install/share/umpire/cmake
install/lib
install/include
install/include/benchmark
install/lib/cmake/benchmark
install/include/umpire
install
install/include/umpire/tpl/json
install/include/umpire/tpl/cxxopts/include
install/include/umpire/tpl/judy
install/include/umpire/resource
install/include/umpire/alloc
install/include/umpire/op
install/include/umpire/util
install/include/umpire/strategy
install/include/umpire/strategy/mixins
install/bin/examples
install/bin/examples/tutorial
install/bin/examples/cookbook
install/bin/tools
```

**Comment by mcfadden8** [comment]
What is `install/lib/cmake/benchmark` for?

**Comment by jdahm** [comment]
Looks like Google Benchmark stuff.

```-- Installing: /home/axom/travis-install/lib/cmake/benchmark/benchmarkConfig.cmake
-- Installing: /home/axom/travis-install/lib/cmake/benchmark/benchmarkConfigVersion.cmake
-- Installing: /home/axom/travis-install/lib/cmake/benchmark/benchmarkTargets.cmake
-- Installing: /home/axom/travis-install/lib/cmake/benchmark/benchmarkTargets-noconfig.cmake
```
You can see the install here: https://dev.azure.com/davidbeckingsale/Umpire/_build/results?buildId=1554

**Comment by davidbeckingsale** [comment]
@jdahm - is there a way to disable installing those benchmark cmake targets?

**Comment by davidbeckingsale** [comment]
Can you also incorporate the changes from #203 in here?

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,6 +40,9 @@ option(ENABLE_COVERAGE "Enable code coverage (with GCC)" Off)
 option(ENABLE_PEDANTIC_WARNINGS "Enable pedantic compiler warnings" On)
 option(ENABLE_TOOLS "Enable Umpire development tools" Off)
 
+# Set benchmark installation to OFF
+set(BENCHMARK_ENABLE_INSTALL OFF)
```
Comment (davidbeckingsale) [inline CMakeLists.txt]: This option will prevent Google Benchmark targets from being installed once BLT contains a newer version of GoogleTest - this is pending.

**Comment by davidbeckingsale** [comment]
@mcfadden8 - regarding the installation of Google Benchmark targets, this PR adds an option that will prevent Google Benchmark targets from being installed, once BLT contains a newer version of GoogleTest. This is pending, and we will create a new PR to update BLT once the GoogleBenchmark version is up to date.

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to
 
 - Generalized Strategy.Device test for all resources.
 
-- Moved `tools/plot_allocations` to `tools/analysis`.
+- Moved `tools/plot_allocations` to `tools/analysis/plot_allocations`.
```
Comment (mcfadden8) [inline CHANGELOG.md]: Is this all that has from this PR?

File: umpire-config.cmake.in [hunk 3]
```diff
@@ -5,6 +5,5 @@
 # SPDX-License-Identifier: (MIT)
 ##############################################################################
 get_filename_component(UMPIRE_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
-set(UMPIRE_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/include")
```
Comment (mcfadden8) [inline umpire-config.cmake.in]: Why did `UMPIRE_INCLUDE_DIRS` get removed?

Comment (jdahm) [reply umpire-config.cmake.in]: Because it seems to have accidentally been removed. I'll put it back!

Comment (mcfadden8) [reply umpire-config.cmake.in]: @jdahm, @davidbeckingsale, now I am wondering how it was working without `UMPIRE_INCLUDE_DIRS` being set.  Are there tests missing that should fail when this cmake variable isn't set?

**Pull request merged**